src/main/java/org/opensearch/searchrelevance/executors/ExperimentRunningManager.java

```diff
@@ -0,0 +1,601 @@
+import org.opensearch.searchrelevance.model.Experiment;
+import org.opensearch.searchrelevance.model.ExperimentType;
+import org.opensearch.searchrelevance.model.QuerySet;
+import org.opensearch.searchrelevance.model.ScheduledExperimentResult;
+import org.opensearch.searchrelevance.model.SearchConfiguration;
+import org.opensearch.searchrelevance.model.SearchConfigurationDetails;
+import org.opensearch.searchrelevance.scheduler.ExperimentCancellationToken;
+import org.opensearch.searchrelevance.settings.SearchRelevanceSettingsAccessor;
+import org.opensearch.searchrelevance.transport.experiment.PutExperimentRequest;
+import org.opensearch.searchrelevance.utils.TimeUtils;
+import org.opensearch.threadpool.ThreadPool;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+/**
+ * ExperimentRunningManager helps isolate the logic for running the steps that was
+ *  in PutExperimentTransportAction and ScheduledExperimentRunnerManager.
+ *
+ * <p>
+ * When running an experiment, we need a {@link QuerySet} and a
+ * list of {@link SearchConfiguration}. We then need to pass that into either
+ * {@link MetricsHelper} for pairwise experiments,
+ * {@link HybridOptimizerExperimentProcessor} for hybrid experiments,
+ * or {@link PointwiseExperimentProcessor}
+ */
+@Log4j2
+@AllArgsConstructor
+public class ExperimentRunningManager {
+    private ExperimentDao experimentDao;
+    private QuerySetDao querySetDao;
+    private SearchConfigurationDao searchConfigurationDao;
+    private ScheduledExperimentHistoryDao scheduledExperimentHistoryDao;
+    private MetricsHelper metricsHelper;
+    private HybridOptimizerExperimentProcessor hybridOptimizerExperimentProcessor;
+    private PointwiseExperimentProcessor pointwiseExperimentProcessor;
+    private ThreadPool threadPool;
+    private SearchRelevanceSettingsAccessor settingsAccessor;
+    private final Map<String, List<Future<?>>> runningFutures = new ConcurrentHashMap<>();
+
+    /**
+     * Starts the experiment by setting up cancellation callback for the cancellation token and also retrieves the queryset
+     * @param experimentId - the id of the experiment to be run
+     * @param request - required parameters for placing a request to start an experiment
+     * @param cancellationToken - reference to cancellation state of scheduled experiment and cancels all futures tied to it
+     * @param actuallyFinished - tracks when the task and all asynchronous processes are completed
+     */
+    public void startExperimentRun(
+        String experimentId,
+        PutExperimentRequest request,
+        ExperimentCancellationToken cancellationToken,
+        CountDownLatch actuallyFinished
+    ) {
+        List<Future<?>> futures = new ArrayList<>();
+        if (request.getScheduledExperimentResultId() != null) {
+            runningFutures.put(request.getScheduledExperimentResultId(), futures);
```

Comment:
with such logic for adding items into the map, the values are not thread-safe, multiple threads can mutate same list concurrently. 
You can do something like this when you're adding to map:
```
runningFutures.compute(scheduledExperimentResultId, (key, existingList) -> {
    List<Future<?>> list = existingList != null ? existingList : Collections.synchronizedList(new ArrayList<>());
    list.add(future);
    return list;
});
```